addons/website_form/static/src/snippets/s_website_form/001.scss

```diff
@@ -54,4 +54,9 @@
             display: none;
         }
     }
+
+    // Used to prevent firefox default red box-shadow on required fields.
+    .s_website_form_input:-moz-ui-invalid:not(:focus) {
```

Comment:
Can't you remove the invalid state right after the `.val('')` call you described instead? That way we'll avoid useless css to be loaded

Reply:
I don't think its related to the validation state since I tried before the ".classList.remove('is-invalid')" workaround and it removed only error style after click on submit (firefox adds a :-moz-ui-invalid in those cases : https://developer.mozilla.org/en-US/docs/Web/CSS/:-moz-ui-invalid)
I'll check if there is another js workaround :)

Reply:
@xO-Tx I did not mean removing the class but the invalid *state*. So removing the :-moz-ui-invalid.